process/transactions/esdtTransactionProcessor.go

```diff
@@ -36,36 +41,149 @@ func (etp *esdtTransactionProcessor) initESDTOperations() {
+	_, esdtFunc := etp.esdtOperations[functionName]
+	_, nftFunc := etp.esdtNFTOperations[functionName]
+
+	return esdtFunc || nftFunc
+}
+
+func (etp *esdtTransactionProcessor) isNFTTx(txData []byte) bool {
+	functionName := etp.getFunctionName(txData)
+	if functionName == "" {
+		return false
+	}
+
+	_, ok := etp.esdtNFTOperations[functionName]
 	return ok
 }
 
-func (etp *esdtTransactionProcessor) getTokenIdentifierAndValue(
-	tx data.TransactionHandler,
-) (tokenIdentifier string, value string) {
-	txData := tx.GetData()
+func (etp *esdtTransactionProcessor) getTokenIdentifier(txData []byte) string {
 	_, arguments, err := etp.argumentParser.ParseData(string(txData))
 	if err != nil {
-		return
+		return ""
 	}
 
 	if len(arguments) >= 1 {
-		tokenIdentifier = string(arguments[0])
+		return string(arguments[0])
 	}
-	if len(arguments) >= 2 {
-		bigValue := big.NewInt(0).SetBytes(arguments[1])
-		value = bigValue.String()
-		if value == "0" {
-			value = ""
-		}
+
+	return ""
+}
+
+func (etp *esdtTransactionProcessor) getNFTTxInfo(txData []byte) (tokenIdentifier, nonce string) {
+	function, arguments, err := etp.argumentParser.ParseData(string(txData))
+	if err != nil {
+		return
+	}
+
+	if len(arguments) < 1 {
+		return
+	}
+
+	_, nftFunc := etp.esdtNFTOperations[function]
+	if !nftFunc {
+		return
+	}
+
+	tokenIdentifier = string(arguments[0])
+
+	if len(arguments) < 2 {
+		return
+	}
+
+	switch function {
+	case core.BuiltInFunctionESDTNFTTransfer, "wipeSingleNFT", core.BuiltInFunctionESDTNFTAddQuantity:
+		nonce = big.NewInt(0).SetBytes(arguments[1]).String()
 	}
 
 	return
 }
+
+func (etp *esdtTransactionProcessor) searchTxsWithNFTCreateAndPutNonceInAlteredAddress(
+	alteredAddresses map[string]*data.AlteredAccount,
+	txs map[string]*data.Transaction,
+	scrs []*data.ScResult,
+) {
+	for txHash, tx := range txs {
+		if !strings.HasPrefix(string(tx.Data), core.BuiltInFunctionESDTNFTCreate) {
+			continue
+		}
+
+		encodedHash := hex.EncodeToString([]byte(txHash))
+		searchSCRWithNonceOfNFT(encodedHash, alteredAddresses, scrs)
+	}
+}
+
+func (etp *esdtTransactionProcessor) searchSCRSWithCreateNFTAndPutNonceInAlteredAddress(
+	alteredAddresses map[string]*data.AlteredAccount,
+	scrs []*data.ScResult,
+) {
+	for _, scr := range scrs {
+		if !strings.HasPrefix(string(scr.Data), core.BuiltInFunctionESDTNFTCreate) {
+			continue
+		}
+
+		searchSCRWithNonceOfNFT(scr.OriginalTxHash, alteredAddresses, scrs)
+	}
+}
+
+func searchSCRWithNonceOfNFT(txHash string, alteredAddresses map[string]*data.AlteredAccount, scrs []*data.ScResult) {
+	for _, scr := range scrs {
+		if scr.OriginalTxHash != txHash {
+			continue
+		}
+
+		if scr.Receiver != scr.Sender {
+			continue
+		}
+
+		altered, ok := alteredAddresses[scr.Sender]
+		if !ok {
+			return
+		}
+
+		altered.NFTNonceString = extractNonceString(scr)
+	}
+}
+
+func extractNonceString(scr *data.ScResult) string {
+	scrDataSplit := bytes.Split(scr.Data, []byte("@"))
```

Comment:
argument parser usage as to not reinvent the wheel in a well hidden place?

Reply:
done. created a new component that extends argument parser

Reply:
great :+1: